readthedocs/projects/views/private.py

```diff
@@ -955,6 +971,80 @@ def get_success_url(self):
         return reverse('projects_integrations', args=[self.get_project().slug])
 
 
+class IntegrationConnectProject(IntegrationMixin, GenericView):
+
+    """
+    Connect a ``Project`` to a ``RemoteRepository``.
+
+    Useful when the Project has been imported manually or got disconnected from
+    the ``RemoteRepository`` instance for some reason.
+
+    Connecting the repository will allow people to handle permissions access
+    permissions (VCS SSO) and also when sending build status notifications,
+    among others.
+    """
+
+    @staticmethod
+    def get_final_url(project):
+        final_url = None
+        url = project.repo
+        if "@" in url:
+            # NOTE: Do we need to convert this? After checking with
+            # community/commercial data it seems it's not required and it will
+            # match with `RemoteRepository.ssh_url`
+            #
+            # url = convert_to_https(url)
+            final_url = url
+        if url.startswith("http"):
+            try:
+                response = requests.head(project.repo, allow_redirects=True)
+                final_url = response.url
+            except requests.exceptions.TooManyRedirects:
+                pass
+        return final_url
+
+    @staticmethod
+    def get_remote_repository(project, url, user):
+        return RemoteRepository.objects.filter(
+            Q(html_url=url) | Q(ssh_url=url),
+            remote_repository_relations__user=user,
+            remote_repository_relations__admin=True,
+        ).first()
+
+    def post(self, request, project_slug, *args, **kwargs):
+        project = self.get_project()
+        url = self.get_final_url(project)
+        remote_repository = self.get_remote_repository(project, url, request.user)
+        log.bind(
+            project_slug=project_slug,
+            repository_url=url,
+            user_username=request.user.username,
+        )
+        if url and remote_repository:
+            log.info(
+                "Project connected to a RemoteRepository",
+                remote_repository_full_name=remote_repository.full_name,
+                remote_repository_id=remote_repository.pk,
+            )
+            project.remote_repository = remote_repository
+            project.save()
+
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                _(f'Project successfully connected to "{url}".'),
+            )
+        else:
+            messages.add_message(
+                request,
+                messages.ERROR,
+                _(f"There was a problem connect your project."),
```

Comment:
```suggestion
                _(f"There was a problem connecting your project."),
```

Maybe we could add more information about the reason? Looks like if there isn't a remote_repository, it means the user isn't an admin or the project hasn't been synced yet.